### PR TITLE
Add facility to take obs from multiple dmpdir cycles, forward and back

### LINKED ISSUE #881: SST converter is failing
I'm not sure what the issue is yet. The ioda converter for sst is segfaulting when ran as part of `test_gdasapp_soca_JGLOBAL_PREP_OCEAN_OB` but runs fine when tested outside ...

### PR DESCRIPTION
**Pull request by AndrewEichmann-NOAA**
Number of cycles forward and back are set in new entries in `parm/soca/obsprep/obsprep_config.yaml`, with a hardcoded default of 0.

More cycles of obs (adt and metop b sst) have been added to the ctests and  `test/soca/testdata/` had been reorganized to a structure more resembling the `dmpdir` in order to keep everything straight.

Most of the "Files changed" are cdl files added to the testdata.

At least partially addresses https://github.com/NOAA-EMC/GDASApp/issues/837

Now also fixes https://github.com/NOAA-EMC/GDASApp/issues/881

Tested with `DMPDIR: /scratch1/NCEPDEV/stmp4/Shastri.Paturi/forAndrew` as well - `JGLOBAL_PREP_OCEAN_OBS` take 16 minutes

### CODE REVIEW

File: ush/soca/prep_marine_obs.py [hunk 1]
```diff
@@ -63,29 +62,38 @@
...
-        obsPath = os.path.join(dataDir, obsSource)
-        obsDestination = os.path.join(COMIN_OBS, obsSource)
-        obsCopy.append([obsPath, obsDestination])
+    for matchingFile in matchingFiles:
+        filePath = os.path.join(matchingFile[0], matchingFile[1])
+        fileDestination = os.path.join(COMIN_OBS, matchingFile[2])
+        fileCopy.append([filePath, fileDestination])
 
-    print(f"obsCopy: {obsCopy}")
+    print(f"fileCopy: {fileCopy}")
```
Comment (guillaumevernieres) [inline ush/soca/prep_marine_obs.py]: since logging is working now ...

File: parm/soca/obsprep/obsprep_config.yaml [hunk 2]
```diff
@@ -39,6 +42,9 @@ observations:
       units: C
       min: -3.0
       max: 50.0
+    window:
```
Comment (guillaumevernieres) [inline parm/soca/obsprep/obsprep_config.yaml:45]: :+1:

**Review by guillaumevernieres — DISMISSED** [review]
Looks good @AndrewEichmann-NOAA , thanks for the quick turn around.

**Review by ShastriPaturi — DISMISSED** [review]

**Review by apchoiCMD — DISMISSED** [review]

**Review by ShastriPaturi — DISMISSED** [review]

File: ush/soca/prep_marine_obs.py (hunk 1 above)
Comment (guillaumevernieres) [reply ush/soca/prep_marine_obs.py]: no need to retest, and since you're on a roll @AndrewEichmann-NOAA , could you swap that to logging pretty please?

Comment (AndrewEichmann-NOAA) [reply ush/soca/prep_marine_obs.py]: @guillaumevernieres done

**Comment by AndrewEichmann-NOAA** [comment]
wait a sec, that last one broke it!

File: ush/soca/prep_marine_obs.py (hunk 1 above)
Comment (AndrewEichmann-NOAA) [reply ush/soca/prep_marine_obs.py]: @guillaumevernieres done and short-tested

**Comment by AndrewEichmann-NOAA** [comment]
ok better now

**Review by guillaumevernieres — APPROVED** [review]
You rock @AndrewEichmann-NOAA ! Thanks for the quick turn around.

**Pull request merged**